examples/migration/recursive_fibonacci/fibonacci_single_task.h

```diff
@@ -46,6 +47,16 @@ struct single_fib_task : task_emulation::base_task {
             }
             case state::sum : {
                 *x = x_l + x_r;
+
+                // Recycling
+                this->s = state::print_status;
+                this->recycle_as_continuation();
+                this->operator()();
+
+                break;
+            }
+            case state::print_status: {
+                std::cout << std::to_string(n) + "\n";
                 break;
             }
```

Comment:
In this example, recycling the task again results in doing `m_parent->remove_child_reference()` after `m_parent` has destroyed itself.